lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2058,6 +2058,11 @@ Instruction *InstCombiner::visitICmpInstWithInstAndIntCst(ICmpInst &ICI,
     // smaller constant, which will be target friendly.
     unsigned Amt = ShAmt->getLimitedValue(TypeBits-1);
     if (LHSI->hasOneUse() &&
+        // @LOCALMOD-BEGIN
+        // We don't want to introduce non-power-of-two integer sizes for PNaCl's
+        // stable wire format, so modify this transformation for NaCl.
+        isPowerOf2_32(TypeBits - Amt) && (TypeBits - Amt) >= 8 &&
```

Comment:
Although this seems suspicious, it also seems like something that we wouldn't hit anyway (we never use e.g. i55) so this appears fine